libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -113,6 +113,7 @@ pair<string, string> IRGenerator::run(
 		solAssert(false, ir + "\n\nInvalid IR generated:\n" + errorMessage + "\n");
 	}
 	asmStack.optimize();
+	asmStack.optimize(); // FIXME: We should be getting good results without running this twice
```

Comment:
This is the place in which I'd rather remove this really... why does this function deal with any of this in the first place :-)? It's an IR generator, it should just, you know, generate IR, shouldn't it :-)?
It's probably solely for the purpose of the ``--ir-optimized`` output... which we may change in the future anyways, since it's horrible confusing (unfortunately breaking, though)...

So I'd suggest to remove this entire mess here, deal with ``--ir-optimized`` output on-demand-only in the compiler stack instead and to only deal with unoptimized IR here, which is then only optimized in ``generateEVMFromIR`` directly on the other side as well.